packages/transit-vehicle-overlay/src/index.story.js

```diff
@@ -4,168 +4,287 @@ import { withA11y } from "@storybook/addon-a11y";
+  }
 
   return (
-    <BaseMap center={portland}>
-      <Vehicles
-        name={name}
-        visible={boolean("show vehicles:", visible)}
-        geometryUrl={geometryUrl}
-        vehicleUrl={vehicleUrl}
-        vehicleQuery={text(
-          "route numbers / all:",
-          "routes/all?or=routes/90,100,190,200,290"
-        )}
-        tracked={text("block -or- trip id:", "track block: 9047")}
-        refreshDelay={number("refresh interval:", 5000)}
-        color={color("color:", "#777")}
-        highlightColor={color("tracked color:", "#ece90d")}
-        lowlightColor={color("trailing color:", "#AAA")}
-        recenterMap={boolean("follow vehicle:", true)}
-        panOffsetX={number("panOffsetX:", 0)}
-        panOffsetY={number("panOffsetY:", 0)}
-        hasPopup={boolean(
-          "use marker popups -- note: edit story to 'false' and refresh:",
-          true
-        )}
-        hasTooltip={boolean("use marker tooltips (desktop only):", true)}
-        onTrackedVehicleUpdate={console.log}
+    <BaseMap
+      center={PORTLAND}
+      zoom={INITIAL_ZOOM_LEVEL}
+      onViewportChanged={onViewportChanged}
+    >
+      <TransitVehicleOverlay
+        zoom={zoom}
+        center={center}
+        vehicleList={vehicleList}
+        onVehicleClicked={clickVehicle}
+        selectedVehicle={tv}
+        showOnlyTracked={showOnlyTracked}
+        pattern={getRoutePattern(tv)}
+        onRecenterMap={recenter}
+        color={clr}
+        highlightColor={highlightColor}
+        MarkerSlot={markers}
+        TooltipSlot={VehicleTooltip}
+        PopupSlot={VehiclePopup}
       />
     </BaseMap>
   );
 }
 
-function rtExampleLayerSwitcher() {
-  return rtExample("Real-Time Buses and Trains", true);
+function simple() {
+  // convert geojson data into expected pattern format
+  const pattern = utils.makePattern(geom, "111");
+  return simpleExample(all, pattern, "9050");
+}
+
+function alternate() {
+  const altVehicles = utils.convertAltData(altLine);
+  const altPattern = utils.makePattern(altGeom, "222");
+  return simpleExample(altVehicles, altPattern, "9088");
+}
+
+function simpleRectangles() {
+  return rectangles(false);
 }
-function rtExampleProgrammatic() {
-  return rtExample(null, false);
+
+function rtCircles() {
+  return realtimeExample(
+    proprietary.fetchVehicles,
+    proprietary.fetchPatternThrottled,
+    ModeCircles
+  );
+}
+
+function rtRectangles() {
+  return realtimeExample(
+    proprietary.fetchAltVehicles,
+    proprietary.fetchPatternThrottled,
+    ModeRectangles
+  );
 }
 
 storiesOf("TransitVehicleOverlay", module)
   .addDecorator(withA11y)
   .addDecorator(withInfo)
   .addDecorator(withKnobs)
-  .add("by Route", routeExample)
-  .add("all Routes", allExample)
-  .add("animated VehicleLayer", animatedExample)
-  .add("real-time vehicles in layer switcher", rtExampleLayerSwitcher)
-  .add("real-time vehicles programmatic control", rtExampleProgrammatic);
+  .add("simple overlay", simple)
```

Comment:
Sometimes the tracked vehicle seems to get stuck. Screenshot shows the simple example that defaults to tracking MAX Blue. I changed to route 57, but the route geometry rendered did not switch over.

![image](https://user-images.githubusercontent.com/2370911/81189274-67506b00-8f84-11ea-9999-2922d709da9d.png)

Reply:
There's just one route geometry (for the rail line) being pulled in from the __mocks__ directory.   The line split still happens nearest the x-axis of the selected vehicle.